src/core.js

```diff
@@ -16,8 +16,8 @@ import {
 import { arrayMap, arrayEach, arrayReduce, getDifferenceOfArrays, stringToArray } from './helpers/array';
 import { instanceToHTML } from './utils/parseTable';
 import { getPlugin } from './plugins';
-import { getRenderer } from './renderers';
-import { getValidator } from './validators';
+import { getRenderer } from './renderers/renderers';
+import { getValidator } from './validators/validators';
```

Comment:
The API is unclear here. I've tried to replace that with
```js
import { getRenderer } from './renderers';
import { getValidator } from './validators';
```
which at first seemed to be better for maintenance (the modules itself exports all necessary API to work with, so there is no reason to import "internal" file - easier refactor in the future). However, it turned out that by accident, I registered all renderers and validators. The unit tests show that something is wrong, but without pointing that I've broken the `handsontable/base` import. I think that the API should be changed. For sure, auto-registering causes complexity.

The same problems occur in other parts of the codebase.

cc @wojciechczerniak

Reply:
So, here are two major issues actually.

1) As you noticed, this complexity seems to be unnecessary. I know we have no so much examples/information in official docs about `/es` and `/commonjs` bundles - but that was possible to import only part of code (BasePlugin/BaseEditor/etc...) and our major goal is to avoid breaking change. 
That complexity of paths and additional reexport in indexes etc will be simplified in the next major release.
@wojciechczerniak noticed, that some end-developers might not have auto resolving files' extensions, so that's the reason to reexport `handsontable/part/asset/index` by `handsontable/part/asset.js` - I agree it's an awful workaround.

2) Auto-registering and default exports are completely killing our dreams about out of the box tree-shaking in end-apps but at the moment implementing the other approach (declarative, I guess), would be another breaking change.

Maybe we could schedule a brainstorm meeting and cast an eye on these issues?

Reply:
> I know we have no so much examples/information in official docs about /es and /commonjs bundles - but that was possible to import only part of code (BasePlugin/BaseEditor/etc...) and our major goal is to avoid breaking change.

In the JS ecosystem, everything is possible. If someone imports something internally (after sniffing files from "node_modules/handsontable") it is their responsibility that something may break in the future. The same with `es/` or `commonjs/` files. The bundler should be responsible for choosing the right module type. 

> @wojciechczerniak noticed, that some end-developers might not have auto resolving files' extensions, so that's the reason to reexport handsontable/part/asset/index by handsontable/part/asset.js - I agree it's an awful workaround.

@wojciechczerniak can you explain what you meant by "not have auto resolving files' extensions"? As far as I know, NodeJS resolves directory access to `index.js`. Maybe I missed something?

> Maybe we could schedule a brainstorm meeting and cast an eye on these issues?

Sure